src/components/scrollManager.js

```diff
@@ -0,0 +1,501 @@
+        }
+
+        return clamp(Math.round(scroll), 0, maxScroll);
+    }
+
+    /**
+     * Calls scrollTo function in proper way.
+     *
+     * @param {HTMLElement} scroller scroller
+     * @param {ScrollToOptions} options scroll options
+     */
+    function scrollToHelper(scroller, options) {
+        if ("scrollTo" in scroller) {
+            if (!supportsScrollToOptions) {
+                var scrollX = (options.left !== undefined ? options.left : scroller.scrollLeft);
+                var scrollY = (options.top !== undefined ? options.top : scroller.scrollTop);
+                scroller.scrollTo(scrollX, scrollY);
+            } else {
+                scroller.scrollTo(options);
+            }
+        } else if ("scrollLeft" in scroller) {
+            if (options.left !== undefined) {
+                scroller.scrollLeft = options.left;
+            }
+            if (options.top !== undefined) {
+                scroller.scrollTop = options.top;
+            }
+        }
+    }
+
+    /**
+     * Performs built-in scroll.
+     *
+     * @param {HTMLElement} xScroller horizontal scroller
+     * @param {number} scrollX horizontal coordinate
+     * @param {HTMLElement} yScroller vertical scroller
+     * @param {number} scrollY vertical coordinate
+     * @param {boolean} smooth smooth scrolling
+     */
+    function builtinScroll(xScroller, scrollX, yScroller, scrollY, smooth) {
+        var scrollBehavior = smooth ? "smooth" : "instant";
+
+        if (xScroller !== yScroller) {
+            scrollToHelper(xScroller, {left: scrollX, behavior: scrollBehavior});
+            scrollToHelper(yScroller, {top: scrollY, behavior: scrollBehavior});
+        } else {
+            scrollToHelper(xScroller, {left: scrollX, top: scrollY, behavior: scrollBehavior});
+        }
+    }
+
+    var scrollTimer;
+
+    /**
+     * Resets scroll timer to stop scrolling.
+     */
+    function resetScrollTimer() {
+        cancelAnimationFrame(scrollTimer);
+        scrollTimer = undefined;
+    }
+
+    /**
+     * Performs animated scroll.
+     *
+     * @param {HTMLElement} xScroller horizontal scroller
+     * @param {number} scrollX horizontal coordinate
+     * @param {HTMLElement} yScroller vertical scroller
+     * @param {number} scrollY vertical coordinate
+     */
+    function animateScroll(xScroller, scrollX, yScroller, scrollY) {
+
+        var ox = xScroller.scrollLeft;
+        var oy = yScroller.scrollTop;
+        var dx = scrollX - ox;
+        var dy = scrollY - oy;
+
+        if (Math.abs(dx) < Epsilon && Math.abs(dy) < Epsilon) {
+            return;
+        }
+
+        var start;
+
+        function scrollAnim(currentTimestamp) {
+
+            start = start || currentTimestamp;
+
+            var k = Math.min(1, (currentTimestamp - start) / ScrollTime);
+
+            if (k === 1) {
+                resetScrollTimer();
+                xScroller.scrollLeft = scrollX;
+                yScroller.scrollTop = scrollY;
```

Comment:
shouldn't this be done by calling `scrollToHelper()`?

Reply:
This completes the scroll if some fractional parts remain.
At that time, I still hadn't decided if I wanted to call the function.
If only for uniformity.
```
builtinScroll(xScroller, scrollX, yScroller, scrollY, false);
```

Reply:
I'm worried about elements not having a setter for e.g. `.scrollLeft`, which is handled by your helper but not by this place in the code